core/src/weighted_shuffle.rs

```diff
@@ -33,6 +33,37 @@ where
         .collect()
 }
 
+pub fn weighted_best(weights: &[(u64, usize)], rng: ChaChaRng) -> usize {
```

Comment:
I'd be nice to have a comment here that documents what this is doing and why it's different